AVIenabledbyArc/vi-edge-deployment-script.sh

```diff
@@ -52,24 +57,51 @@ function create_cognitive_hobo_resources {
   result=$(az rest --method post --uri $createResourceUri 2>&1 >/dev/null || true)
   echo $result    
 
-  if [[ "$result" == *"ERROR:"* ]]; then
-    echo "CS Resources already exist. ignoring"
+  if [[ "$result" == *"ERROR: Conflict"* ]]; then
+    echo "CS Resources already exist. Moving on."
   else
-    echo "sleeping for $sleepDuration seconds"  
+    echo "No CS resources found. Creating"  
+  fi
+  getSecretsUri="https://management.azure.com/subscriptions/${viSubscriptionId}/resourceGroups/${viResourceGroup}/providers/Microsoft.VideoIndexer/accounts/${viAccountName}/ListExtensionDependenciesData?api-version=${viApiVersion}"
+  csResourcesData=$(az rest --method post --uri $getSecretsUri 2>&1 >/dev/null || true)
+  if [[ "$csResourcesData" == *"ERROR"* ]]; then
+    printf "\n"
+    numRetries=0
+    sleepDuration=1
+    maxNumRetries=20
+    while  [ $numRetries -lt $maxNumRetries ]; do
+      csResourcesData=$(az rest --method post --uri $getSecretsUri 2>&1 >/dev/null || true)
+      if [[ "$csResourcesData" == *"ERROR"* ]]; then
+          numRetries=$(( $numRetries + 1 ))
+          progress=$(( $numRetries*100/20 ))
+          progress-bar 100 $progress; 
     sleep $sleepDuration
+      else
+          progress-bar 100 100; 
+          break
+      fi
+    done
+      printf "\n"
   fi
-  echo "CS resources has been created"
 
   echo "=============================="
   echo "Getting secrets"
   echo "=============================="
-  getSecretsUri="https://management.azure.com/subscriptions/${viSubscriptionId}/resourceGroups/${viResourceGroup}/providers/Microsoft.VideoIndexer/accounts/${viAccountName}/ListExtensionDependenciesData?api-version=${viApiVersion}"
+
+  printf "\n"
+  if [[ "$csResourcesData" == *"ERROR:"* ]]; then
+    echo "Error getting the cognitive services resources, please reach out to support"
```

Comment:
Shouldn't you break  / throw?

Reply:
no , if the resource does not exists , an error wil lbe returned but we also provision new Hobo behind the scenes and we need to wait for it to finish . that is why there is a loop under this code : 

![image](https://github.com/Azure-Samples/media-services-video-indexer/assets/512709/dfc8440f-6192-418d-b833-55980b182e54)

Reply:
you are right . the code was chaged a lot please review it again